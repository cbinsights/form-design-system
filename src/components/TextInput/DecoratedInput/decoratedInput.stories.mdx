import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import { withKnobs, text } from '@storybook/addon-knobs';
import { ImportPath } from 'components/util/storybook';

import Flex from 'components/Flex';
import FlexItem from 'components/FlexItem';
import DecoratedInput from '.';

<Meta title="Components/DecoratedInput" decorators={[withKnobs]} component={DecoratedInput} />

# DecoratedInput

<ImportPath component={DecoratedInput} section="forms" />

Wrapper for `input` elements to add contextual labels to either side of the
input. **Side labels are not a substitute for `label` elements!**

<Story name="Knobs" parameters={{ docs: { disable: true }}}>
  <React.Fragment>
    <p className="margin--bottom type--data color--purple fontStyle--italic">
      Use the knobs panel to change left and right labels
    </p>
    <label htmlFor="horseName">Name your horse</label>
    <DecoratedInput
      before={text('before', 'before')}
      after={text('after', 'after')}
    >
      <input id="horseName" type="text" />
    </DecoratedInput>
  </React.Fragment>
</Story>

<Preview isExpanded>
  <React.Fragment>
    <label htmlFor="horseName">Name your horse</label>
    <DecoratedInput
      before='before'
      after='after'
    >
      <input id="horseName" type="text" />
    </DecoratedInput>
  </React.Fragment>
</Preview>

<Props of={DecoratedInput} />

### Before

<Preview>
  <Story name="Before">
    <DecoratedInput before="£">
      <input type="number" placeholder="Price of lorry" step="0.01" min="0.01" />
    </DecoratedInput>
  </Story>
</Preview>

### After

<Preview>
  <Story name="After">
    <DecoratedInput after="GBP">
      <input type="number" placeholder="Price of lorry" step="0.01" min="0.01" />
    </DecoratedInput>
  </Story>
</Preview>

### Before and After

<Preview>
  <Story name="Before and After">
    <DecoratedInput before="£" after="GBP">
      <input type="number" placeholder="Price of lorry" step="0.01" min="0.01" />
    </DecoratedInput>
  </Story>
</Preview>

### None

<Preview>
  <Story name="None">
    <DecoratedInput>
      <input type="number" placeholder="Price of lorry" step="0.01" min="0.01" />
    </DecoratedInput>
  </Story>
</Preview>