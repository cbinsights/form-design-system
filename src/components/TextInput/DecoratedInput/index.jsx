import React from 'react';
import PropTypes from 'prop-types';

/**
 * @param {Object} props react props
 * @returns {ReactElement}
 */
const DecoratedInput = ({ before, after, children }) => (
  <div className="fdsDecoratedInput" data-before={before} data-after={after}>
    {children}
  </div>
);

DecoratedInput.propTypes = {
  /** String to place to the left of the input */
  before: PropTypes.string,

  /** String to place to the right of the input */
  after: PropTypes.string,

  /** React children */
  children: PropTypes.node,
};

export default DecoratedInput;
