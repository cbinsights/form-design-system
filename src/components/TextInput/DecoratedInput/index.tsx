import React from 'react';

export interface DecoratedInputProps {
  /** String to place to the left of the input */
  before?: string;

  /** String to place to the right of the input */
  after?: string;

  /** React children */
  children?: Array<React.ReactNode> | React.ReactNode;
}

const DecoratedInput = ({
  before,
  after,
  children,
}: DecoratedInputProps): JSX.Element => (
  <div className="fdsDecoratedInput" data-before={before} data-after={after}>
    {children}
  </div>
);

export default DecoratedInput;
