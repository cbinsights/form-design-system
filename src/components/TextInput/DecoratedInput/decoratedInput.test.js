import React from 'react';
import { shallow } from 'enzyme';

import DecoratedInput from '.';

describe('DecoratedInput component', () => {
  it('matches snapshot (set all props)', () => {
    const wrapper = shallow(
      <DecoratedInput
        before="left boi"
        after="right boi"
      >
        <input type="text" />
      </DecoratedInput>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('does NOT add data attribute for before if undefined', () => {
    const wrapper = shallow(
      <DecoratedInput
        after="right boi"
      >
        <input type="text" />
      </DecoratedInput>
    );
    expect(wrapper.exists('[data-after]')).toBe(true);
    expect(wrapper.exists('[data-before]')).toBe(false);
  });

  it('does NOT add data attribute for after if undefined', () => {
    const wrapper = shallow(
      <DecoratedInput
        before="left boi"
      >
        <input type="text" />
      </DecoratedInput>
    );
    expect(wrapper.exists('[data-after]')).toBe(false);
    expect(wrapper.exists('[data-before]')).toBe(true);
  });
});
