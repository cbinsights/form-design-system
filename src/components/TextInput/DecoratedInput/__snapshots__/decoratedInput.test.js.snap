// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DecoratedInput component matches snapshot (set all props) 1`] = `
<div
  className="fdsDecoratedInput"
  data-after="right boi"
  data-before="left boi"
>
  <input
    type="text"
  />
</div>
`;
